### ISSUE TITLE
remote-exec stops executing after running `sudo systemctl start`

### ISSUE THREAD

**Issue by paulczar** [hashicorp/terraform#5737]
This applys to the current release `v0.6.13`...  `v0.6.8` works as expected.

This means only the first service that I ask for actually starts in the current version, whereas the earlier version it all starts up as expected.

## Code Snippet

The following block in remote-exec:

```
    provisioner "remote-exec" {
        inline = [
            "echo '==> Services'",
            "echo '----> starting registrator'",
            "sudo systemctl start registrator.service",
            "echo '----> starting swarm-agent'",
            "sudo systemctl start swarm-agent.service",
            "echo '----> starting swarm-manager'",
            "sudo systemctl start swarm-manager.service",
        ]
    }
```

## Run output - v0.6.13

```
null_resource.install_swarm.0 (remote-exec): ==> Services
null_resource.install_swarm.0 (remote-exec): ----> starting registrator
null_resource.install_swarm.2 (remote-exec): ==> Services
null_resource.install_swarm.2 (remote-exec): ----> starting registrator
null_resource.install_swarm.1 (remote-exec): ==> Services
null_resource.install_swarm.1 (remote-exec): ----> starting registrator
null_resource.install_swarm.2: Creation complete
null_resource.install_swarm.0: Creation complete
null_resource.install_swarm.1: Creation complete
```
## Run output - v0.6.8

```
null_resource.install_swarm.1 (remote-exec): ==> Services
null_resource.install_swarm.1 (remote-exec): ----> starting registrator
null_resource.install_swarm.2 (remote-exec): ==> Services
null_resource.install_swarm.2 (remote-exec): ----> starting registrator
null_resource.install_swarm.0 (remote-exec): ==> Services
null_resource.install_swarm.0 (remote-exec): ----> starting registrator
null_resource.install_swarm.1 (remote-exec): ----> starting swarm-agent
null_resource.install_swarm.0 (remote-exec): ----> starting swarm-agent
null_resource.install_swarm.2 (remote-exec): ----> starting swarm-agent
null_resource.install_swarm.1 (remote-exec): ----> starting swarm-manager
null_resource.install_swarm.0 (remote-exec): ----> starting swarm-manager
null_resource.install_swarm.2 (remote-exec): ----> starting swarm-manager
null_resource.install_swarm.1: Creation complete
null_resource.install_swarm.2: Creation complete
null_resource.install_swarm.0: Creation complete
```

**Comment by byrnedo** [comment]
Related to #5699

**Comment by phinze** [comment]
Hi @paulczar - thanks for the report and sorry for the trouble here! Looks like the attempt to clean up remote exec scripts in v0.6.13 had an unexpected race condition. Fixed on master and will be fixed in 0.6.14 to be released layer today.
